src/commands/clone.js

```diff
@@ -41,10 +42,33 @@ function clone (opts) {
   var onExit = require('../lib/exit')
   var parseArgs = require('../parse-args')
   var debug = require('debug')('dat')
-
   var parsed = parseArgs(opts)
+
   opts.key = parsed.key || opts._[0] // pass other links to resolver
   opts.dir = parsed.dir
+
+  // variables needed to check if opts.key is on system and leads to a dat.json file.
+  var datPath = opts.key
+  var onSystem = fs.existsSync(datPath)
+  var isDir = false
+  var isDatJson = false
+
+  // if the file or directory is on system check to see whether the path is to a dat.json file or directory.
+  if (onSystem) {
+    isDir = fs.lstatSync(datPath).isDirectory()
+    isDatJson = (fs.lstatSync(datPath).isFile() && (path.basename(datPath) === 'dat.json'))
+  }
+
+  // if file is a dat.json resolve absolute path. else if directory try to resolve path to dat.json.
+  if (isDatJson) {
+    datPath = path.resolve(datPath)
+  } else if (isDir) {
+    datPath = path.join(datPath + '/dat.json')
```

Comment:
@joehand @karissa  The test - `specify directory containing dat.json` breaks if `/dat.json` is changed to `dat.json`.